### PR TITLE
Add MTE-415 [v112] Added Recently Saved smoke tests

### PR DESCRIPTION
**Pull request by SimonBasca**
- Automated tests for https://testrail.stage.mozaws.net/index.php?/cases/view/1485032

### CODE REVIEW

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Edited 6 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 3b6e79a99166c1c2b1a31a1f54f96c1bb61d7631
</p>

**Comment by clarmso** [comment]
👀 Let me take a look on my end.

File: Tests/XCUITests/BaseTestCase.swift [hunk 1]
```diff
@@ -112,6 +114,78 @@ class BaseTestCase: XCTestCase {
...
+        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
+        waitUntilPageLoad()
+        waitForExistence(app.buttons["Reader View"], timeout: TIMEOUT)
+        app.buttons["Reader View"].tap()
+        waitUntilPageLoad()
+        waitForExistence(app.buttons["Add to Reading List"])
+        app.buttons["Add to Reading List"].tap()
+    }
+    func removeContentFromReaderView() {
+        navigator.goto(LibraryPanel_ReadingList)
```
Comment (clarmso) [inline Tests/XCUITests/BaseTestCase.swift:173]: The test fails on both iPhone and iPad here on my end.

File: Tests/XCUITests/HomePageSettingsUITest.swift [hunk 2]
```diff
@@ -221,6 +222,44 @@ class HomePageSettingsUITests: BaseTestCase {
...
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        checkRecentlySaved()
+        app.scrollViews.cells[AccessibilityIdentifiers.FirefoxHomepage.RecentlySaved.itemCell].staticTexts[urlExampleLabel].tap()
+        navigator.nowAt(BrowserTab)
+        waitForTabsButton()
+        unbookmark()
+        if !iPad() {
+            navigator.performAction(Action.CloseTab)
+        }
+        removeContentFromReaderView()
```
Comment (clarmso) [inline Tests/XCUITests/HomePageSettingsUITest.swift:252]: The test fails on my end on both iPhone and iPad here.

**Comment by isabelrios** [comment]
@SimonBasca if I'm not mistaken the test is enabled in both the Full Functional and the Smoketest3, it should be enabled only in the latter. Thanks!
Also, I can reproduce locally the same error as we are seeing on Bitrise. I will ping you on slack

**Review by isabelrios — APPROVED** [review]
LGTM!

**Comment by clarmso** [comment]
The failing test from `Smoketest2` has been passing on main. This is the expected screenshot from the particular line.

Simulator Screen Shot - iPhone 14 - 2023-03-13 at 11 20 15

Let me take a look in this branch. 👀

**Comment by clarmso** [comment]
The failing test from `Smoketest2` is due to a known intermittent issue on the app launch. 🤷🏼 Otherwise LGTM and ok to merge. 
Screenshot_1_9DB3FF6A-A508-41F3-A265-74C671501D0C

**Review by clarmso — APPROVED** [review]

**Pull request merged**